### DISCUSSION TITLE
Why can't I set default props for prebuild components? What is the recommended way to do this?

### DISCUSSION THREAD

**Discussion by ishowta** [chakra-ui/chakra-ui#6025] [Q&A]
IMO, setting default props is a common use case for customization.
However, `defaultProps` in `extendTheme` only accepts some special setting items such as `size` and `variant`. Why is this? Is there some technical problem?

```ts
extendTheme( {
  components: {
    Tooltip: {
      defaultProps: {
        openDelay: 500, // <- not work
      },
    },
  }
})
```

It seems redundant to create a wrapped component like bellow just to set default props.

```tsx
export const Tooltip: FC<TooltipProps> = ({ children, ...props }) => {
  return (
    <ChakraTooltip openDelay={500} {...props}>
      {children}
    </ChakraTooltip>
  )
}
```

Sorry for the basic question.

**Comment by mrmartineau** [comment]
I am also trying to find the correct solution for this. `defaultProps` only allows setting `size`, `variant` or `colorScheme` props, not others like `isRound` on the `IconButton` or `loadingText` on the `Button` component.

**Comment by SamyZog** [comment]
What I ended up doing is 

```
import { Tooltip } from "@chakra-ui/react";

Tooltip.defaultProps = {
   placement: "top"
};
```

**Reply by JunlinPan-Crypto** [reply]
> What I ended up doing is
> 
> ```
> import { Tooltip } from "@chakra-ui/react";
> 
> Tooltip.defaultProps = {
>    placement: "top"
> };
> ```

> Warning: Menu: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.

So, what's the alternative?

**Comment by yarapolana** [comment]
Found the same problem, that's rather unfortunate if you can't set defaultProps when extending the theme.

**Comment by veloware** [comment]
What is the recommended way to do this given `defaultProps` is to be deprecated in react 18.3?

**Comment by aladinflux** [comment]
This is very unfortunate that we can't set some default props like `placement` or `hasArrow` for `Tooltip`
